net/src/net_common.cpp

```diff
@@ -54,3 +57,12 @@ int convert_mbed_addr_to_bsd(struct sockaddr * out, const SocketAddress * in)
 
     return 0;
 }
+
+NetworkInterface * get_mbed_net_if()
+{
+    if (mbed_net_if == nullptr)
```

Comment:
This is not threasafe. I would suggest the following: 
```c++
static NetworkInterface * mbed_net_if = NetworkInterface::get_default_instance();
return mbed_net_if;
```

It is guaranteed to be thread safe with C++11.